src/Network/Wai/SAML2/EntityDescriptor.hs

```diff
@@ -0,0 +1,91 @@
+{-# LANGUAGE LambdaCase #-}
+--------------------------------------------------------------------------------
+-- SAML2 Middleware for WAI                                                   --
+--------------------------------------------------------------------------------
+-- This source code is licensed under the MIT license found in the LICENSE    --
+-- file in the root directory of this source tree.                            --
+--------------------------------------------------------------------------------
+
+module Network.Wai.SAML2.EntityDescriptor
+    ( IDPSSODescriptor(..)
+    , Binding(..)
+    ) where
+
+import qualified Crypto.Store.X509 as X509
+import qualified Data.X509 as X509
+import Data.Text (Text)
+import qualified Data.Map as Map
+import qualified Data.Text as T
+import qualified Data.Text.Encoding as T
+import Network.Wai.SAML2.XML
+import Text.XML.Cursor
+
+-- | Datatype describing metadata of an identity provider.
+-- See also section 2.4.3 of [Metadata for the OASIS Security Assertion Markup Language (SAML) V2.0](https://docs.oasis-open.org/security/saml/v2.0/saml-metadata-2.0-os.pdf).
+data IDPSSODescriptor
+    = IDPSSODescriptor {
+        -- | IdP Entity ID. 'Network.Wai.SAML2.Config.saml2ExpectedIssuer' should be compared against this identifier
+        entityID :: Text
+        -- | The X.509 certificate for signed assertions
+    ,   x509Certificate :: X509.SignedExact X509.Certificate
+        -- | Supported NameID formats
+    ,   nameIDFormats :: [Text]
+        -- | List of SSO urls corresponding to 'Binding's
+    ,   singleSignOnServices :: Map.Map Binding Text
+    } deriving Show
+
+-- | urn:oasis:names:tc:SAML:2.0:bindings
+-- https://docs.oasis-open.org/security/saml/v2.0/saml-bindings-2.0-os.pdf
+data Binding
+    = HTTPPost
+    | HTTPRedirect
+    | HTTPArtifact
+    | PAOS
+    | SOAP
+    | URLEncodingDEFLATE
+    deriving (Show, Eq, Ord)
```

Comment:
Does the `Ord` instance make sense here? (I know it is currently needed because it is used as a key for `Map`, but see my comment there.)